pkg/cmd/pulumi/config.go

```diff
@@ -684,10 +706,88 @@ func newConfigSetAllCmd(stack *string) *cobra.Command {
 	setCmd.PersistentFlags().StringArrayVar(
 		&secretArgs, "secret", []string{},
 		"Marks a value as secret to be encrypted")
+	setCmd.PersistentFlags().StringVar(
+		&rawJsonArgs, "json", "",
+		"Parse the keys as from a json input")
 
 	return setCmd
 }
 
+func parseJsonConfig(jsonConfig string) (map[string]configValueJSON, error) {
+	config := make(map[string]configValueJSON)
+	err := json.Unmarshal([]byte(jsonConfig), &config)
+	if err != nil {
+		return config, err
+	}
+	return config, nil
+}
+
+func encryptValue(ctx context.Context, stack backend.Stack, value string) (string, error) {
+	c, cerr := getStackEncrypter(stack)
+	if cerr != nil {
+		return "", cerr
+	}
+	enc, eerr := c.EncryptValue(ctx, value)
+	if eerr != nil {
+		return "", eerr
+	}
+	return enc, nil
+}
+
+func setKeyValuePairFromJson(ctx context.Context, stack backend.Stack, ps *workspace.ProjectStack, key config.Key, jsonValue configValueJSON) error {
+	var value config.Value
+	if jsonValue.ObjectValue != nil {
+		if jsonValue.Secret {
+			enc, err := encryptValue(ctx, stack, *jsonValue.Value)
+			if err != nil {
+				return err
+			}
+			value = config.NewSecureObjectValue(enc)
+		} else {
+			value = config.NewObjectValue(*jsonValue.Value)
+		}
+		err := ps.Config.Set(key, value, true)
```

Comment:
Could probably lift the Set calls out of this method to `setConfigFromJsonValue` and just have this be a mapping function from jsonVallue to config.Value.